lib/mwUtil.js

```diff
@@ -636,4 +644,18 @@ mwUtil.addVaryHeader = (res, vary) => {
     }
 };
 
+mwUtil.cloneRequest = (req) => {
+    if (!req) {
+        return;
+    }
+    return {
+        method: req.method,
+        uri: req.uri,
+        headers: Object.assign({}, req.headers),
+        query: req.query,
+        body: req.body,
+        params: req.params
+    };
```

Comment:
If we're cloning - let's clone, deep clone. Or for performance we can provide an optional parameter with an array that would specify which portions of the request to clone